bolt04/onion-test-multi-frame.json

```diff
@@ -0,0 +1,42 @@
+{
+  "comment": "A testcase for a variable length hop_payload. The third payload is 256 bytes long.",
+  "generate": {
+    "session_key": "4141414141414141414141414141414141414141414141414141414141414141",
+    "associated_data": "4242424242424242424242424242424242424242424242424242424242424242",
+    "hops": [
+	{
+	    "type": "legacy",
+	    "pubkey": "02eec7245d6b7d2ccb30380bfbe2a3648cd7a942653f5aa340edcea1f283686619",
+	    "payload": "0000000000000000000000000000000000000000"
+	},
+	{
+	    "type": "tlv",
+	    "pubkey": "0324653eac434488002cc06bbfb7f10fe18991e35f9fe4302dbea6d2353dc0ab1c",
+	    "payload": "0101010101010101000000000000000100000001"
+	},
+	{
+	    "type": "raw",
```

Comment:
What's a "raw" payload? Does this mean don't apply any serialization and use it as is? Seems like a entirely new payload type not defined in this PR.

Reply:
Yes this means this is raw bytes and you only need to prepend the length to use it as payload.
My tests end up prepending `0xfd0001` (length 256) to complete the payload and it results in the serialized onion specified in this file.
I agree though that these `types` aren't specified and it may be helpful to detail a bit more the JSON schema.

Reply:
Their only intended to be used for testing random payloads while we narrow down on the TLV fomat that we actually use. The `raw` type is currently the only format that allows > 32 bytes payloads, that's why I added it. It isn't handled by implementations at all since we now cleanly separate payload parsing from onion processing :wink:

If desired I can switch to TLV payloads in another PR, once we have merged both this and finalized the TLV format, but I think having this is nice to test corner cases as well.

Reply:
> Their only intended to be used for testing random payloads while we narrow down on the TLV fomat that we actually use. The raw type is currently the only format that allows > 32 bytes payloads, that's why I added it.

So it's serialized with the same varint length prefix as the TLV payloads?

Reply:
Exactly. You need to add the length prefix before passing this on to your onion processing layer.